client/transaction-pool/graph/src/validated_pool.rs

```diff
@@ -521,14 +529,35 @@ impl<B: ChainApi> ValidatedPool<B> {
 	pub fn status(&self) -> PoolStatus {
 		self.pool.read().status()
 	}
+
+	/// Notify all watchers that transactions in the block with hash have been finalized
+	pub async fn on_block_finalized(&self, block_hash: BlockHash<B>) -> Result<(), B::Error> {
+		debug!(target: "txpool", "Attempting to notify watchers of finalization for {}", block_hash);
+		// fetch all extrinsic hashes
+		let tx_hashes = self.api.block_body(&BlockId::Hash(block_hash.clone())).await?
+			.expect("fetched block header should have block body; qed")
```

Comment:
This proof is most likely valid, but I'd prefer to be on the safe side here and avoid crashing if we ever end up with a hash that does not have a body in DB (for whatever reason).

Could you change that to `if let`?